esphome/components/as7341/sensor.py

```diff
@@ -0,0 +1,113 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import i2c, sensor
+from esphome.const import (
+    CONF_GAIN,
+    CONF_ID,
+    DEVICE_CLASS_ILLUMINANCE,
+    ICON_BRIGHTNESS_5,
+    STATE_CLASS_MEASUREMENT,
+)
+
+
+CODEOWNERS = ["@mrgnr"]
+DEPENDENCIES = ["i2c"]
+
+as7341_ns = cg.esphome_ns.namespace("as7341")
+
+AS7341Component = as7341_ns.class_(
+    "AS7341Component", cg.PollingComponent, i2c.I2CDevice
+)
+
+CONF_ATIME = "atime"
+CONF_ASTEP = "astep"
+
+CONF_F1 = "f1"
+CONF_F2 = "f2"
+CONF_F3 = "f3"
+CONF_F4 = "f4"
+CONF_F5 = "f5"
+CONF_F6 = "f6"
+CONF_F7 = "f7"
+CONF_F8 = "f8"
+CONF_CLEAR = "clear"
+CONF_NIR = "nir"
+
+UNIT_COUNTS = "#"
+
+AS7341_GAIN = as7341_ns.enum("as7341_gain_t")
+GAIN_OPTIONS = {
+    "X0.5": AS7341_GAIN.AS7341_GAIN_0_5X,
+    "X1": AS7341_GAIN.AS7341_GAIN_1X,
+    "X2": AS7341_GAIN.AS7341_GAIN_2X,
+    "X4": AS7341_GAIN.AS7341_GAIN_4X,
+    "X8": AS7341_GAIN.AS7341_GAIN_8X,
+    "X16": AS7341_GAIN.AS7341_GAIN_16X,
+    "X32": AS7341_GAIN.AS7341_GAIN_32X,
+    "X64": AS7341_GAIN.AS7341_GAIN_64X,
+    "X128": AS7341_GAIN.AS7341_GAIN_128X,
+    "X256": AS7341_GAIN.AS7341_GAIN_256X,
+    "X512": AS7341_GAIN.AS7341_GAIN_512X,
+}
+
+
+def sensor_schema():
+    return sensor.sensor_schema(
+        unit_of_measurement=UNIT_COUNTS,
+        icon=ICON_BRIGHTNESS_5,
+        accuracy_decimals=0,
+        device_class=DEVICE_CLASS_ILLUMINANCE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    )
```

Comment:
No need to wrap in a function
```suggestion
SENSOR_SCHEMA = sensor.sensor_schema(
    unit_of_measurement=UNIT_COUNTS,
    icon=ICON_BRIGHTNESS_5,
    accuracy_decimals=0,
    device_class=DEVICE_CLASS_ILLUMINANCE,
    state_class=STATE_CLASS_MEASUREMENT,
)
```